crates/libcontainer/src/workload/mod.rs

```diff
@@ -60,6 +71,16 @@ pub trait Executor: CloneBoxExecutor {
     /// namespace and cgroups, and pivot_root into the rootfs. But this step
     /// runs before waiting for the container start signal.
     fn validate(&self, spec: &Spec) -> Result<(), ExecutorValidationError>;
+
+    /// Set environment variables for the container process to be executed.
+    fn set_envs(&self, envs: HashMap<String, String>) -> Result<(), ExecutorSetEnvsError> {
```

Comment:
I don't think `set` is a proper word for this method because we need to delete and add the envs. How about `update` or `setup`?

Reply:
`setup_envs` sounds good to me. Let me rename the method and related type names.